vispy/visuals/pseudo_spheres.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+"""Pseudo spheres (shaded discs) visual"""
+
+import numpy as np
+
+from ..color import ColorArray
+from ..gloo import VertexBuffer
+from .visual import Visual
+
+
+vert = """
+uniform vec3 u_light_position;
+
+attribute vec3 a_position;
+attribute vec4 a_color;
+attribute float a_radius;
+
+varying vec4 v_color;
+varying vec3 v_light_direction;
+varying float v_depth_middle;
+
+float big_float = 1e10; // prevents wobbling of radii
+
+void main (void) {
+    v_color = a_color;
+    v_light_direction = normalize(u_light_position);
+
+    vec4 pos = vec4(a_position, 1.0);
+    vec4 fb_pos = $visual_to_framebuffer(pos);
+    gl_Position = $framebuffer_to_render(fb_pos);
+
+    // NOTE: gl_stuff uses framebuffer coords!
+
+    // calculate point size from visual to framebuffer coords to determine radius
+    vec4 x = $framebuffer_to_visual(fb_pos + vec4(big_float, 0, 0, 0));
+    x = (x/x.w - pos) / big_float;
+    vec4 radius_vec = $visual_to_framebuffer(pos + normalize(x) * a_radius);
+    float radius = radius_vec.x/radius_vec.w - fb_pos.x/fb_pos.w;
+
+    // gl_PointSize is the diameter
+    gl_PointSize = radius * 2;
+
+    // Get the framebuffer z direction relative to this sphere in visual coords
+    vec4 z = $framebuffer_to_visual(fb_pos + vec4(0, 0, big_float, 0));
+    z = (z/z.w - pos) / big_float;
```

Comment:
is the divide by big_float necessary here? It seems like that should get taken care of by the normalize in the following line (same for the `x = ...` above). I could see this causing some issues if the components end up being small in visual coordinates.

Maybe try removing the divides?

Reply:
Nice catch! It's definitely unnecessary, but unfortunately does not solve the depth issue with `fov != 0` :/